custom_from.php

```diff
@@ -134,10 +260,97 @@ public function message_compose($params)
                         $score = 1;
                     }
                 }
-            }
+
+                if ($score > 2 && !$should_override_identity) {
+                    break;
+                }
+
+                /** Remove autocomplete by JS frontend after page loaded */
+                $rcmail->output->set_env('identities', array());
+
+                break;
+            case self::DEFAULT_IDENTITY_OPTION:
+                $default_identity = $rcmail->user->get_identity();
+                $field_attrib = ['name' => '_from'];
+
+                foreach ($attrib as $attr => $value) {
+                    if (in_array($attr, ['id', 'class', 'style', 'size', 'tabindex'])) {
+                        $field_attrib[$attr] = $value;
+                    }
+                }
+
+                $field_attrib['onchange'] = rcmail_output::JS_OBJECT_NAME . ".change_identity(this)";
+                $select_from = new html_select($field_attrib);
+
+                $select_from->add(format_email_recipient($default_identity['email'], $default_identity['name']), $default_identity['identity_id']);
+
+                $attrib['content'] = $select_from->show($default_identity['identity_id']);;
         }
 
         self::set_state($compose_id, $address);
+
+        return $attrib;
+    }
+
+    /**
+     * Override headers fields from identity (bcc, reply-to), by plugin settings
+     */
+    public function compose_additional_headers($compose_id, $msg_uid, $reply_from, $attrib)
+    {
+        if (!self::get_state($compose_id) && $reply_from !== self::RECEIVING_EMAIL_WITH_DEFAULT_IDENTITY_OPTION) {
+            return $attrib;
+        }
+
+        $part = strtolower($attrib['part']);
+        $value = '';
+
+        switch ($reply_from) {
+            case self::RECEIVING_EMAIL_WITH_DEFAULT_IDENTITY_OPTION:
+                $default_identity = rcmail::get_instance()->user->get_identity();
+                $value = $default_identity[$part] ?? '';
+                $field_attrib = array();
+
+                if (!$value && $part == 'replyto') {
+                    $value = $default_identity['reply-to'] ?? '';
+                }
+            case self::RECEIVING_EMAIL_OPTION:
+                $fname  = '_' . $part;
+                $allow_attrib = ['id', 'class', 'style', 'cols', 'rows', 'tabindex', 'data-recipient-input'];
+
+                foreach ($allow_attrib as $attr) {
+                    if (isset($attrib[$attr])) {
+                        $field_attrib[$attr] = $attrib[$attr];
+                    }
+                }
+
+                // Create a textarea field for overriding Roundcube input fields
+                $field = new html_textarea();
+
+                $attrib['content'] = $field->show($value, ['name' => $fname] + $field_attrib);
```

Comment:
I'm a bit lost on the whole `compose_additional_headers` / `compose_draft_from_headers` / `compose_from_headers` and the updated README file unfortunately didn't allow me to fully understand how this works.

For example this function is taking this `$attrib` array (I don't know what its expected contents is, maybe a code comment or pointer to documentation would help) and it seems the only effect it has is this line modifying its `'content'` key. When entering the function with `$reply_from == self::RECEIVING_EMAIL_OPTION` then we would skip part of the switch, for example `$field_attrib`would not get assigned to any value, yet we try to modify it on line 322.

Some fragments of the change you sent are a bit obscure to me, and the PR being quite large doesn't really help. I would appreciate if you could clarify to me how these new options are expected to work, then maybe I'll be able to suggest another wording for the `README.md` file as I guess if I wasn't able to understand it then it's possible other users would face the same issue.